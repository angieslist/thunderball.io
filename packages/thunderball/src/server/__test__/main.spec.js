import request from 'supertest';

describe('main - app', () => {
  let server;

  beforeEach(() => {
    server = require('../main');
  });

  afterEach(() => {
    server.close();
  });

  it('sends 404 response for nonexistent route', async () => {
    const response = await request(server).get('/not-a-page');
    expect(response).toHaveProperty('status', 404);
    expect(JSON.parse(response.text)).toMatchSnapshot();
  });

  it('sets response headers', async () => {
    const response = await request(server).get('/not-a-page');
    expect(response.headers).toHaveProperty('content-type', 'application/json; charset=utf-8');
    expect(response.headers).toHaveProperty('x-powered-by', 'Thunderball');
    expect(response.headers).toHaveProperty('etag');
  });
});
