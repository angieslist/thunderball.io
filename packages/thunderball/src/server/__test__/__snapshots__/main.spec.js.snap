// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main - app sends 404 response for nonexistent route 1`] = `
Object {
  "error": "Resource not found",
}
`;
